core/src/main/java/org/infinispan/configuration/parsing/SecurityActions.java

```diff
@@ -16,16 +15,7 @@
  * @since 10.0
  */
 final class SecurityActions {
-
-   private static <T> T doPrivileged(PrivilegedAction<T> action) {
-      if (System.getSecurityManager() != null) {
-         return AccessController.doPrivileged(action);
-      } else {
-         return Security.doPrivileged(action);
-      }
-   }
-
    static Properties getSystemProperties() {
-      return doPrivileged(System::getProperties);
+      return doPrivileged((PrivilegedAction<Properties>) System::getProperties);
```

Comment:
I think `doPrivileged` is not required here (it is a no-op and invokes the action directly).